Drawsana/Shapes/Implementations/PenShape.swift

```diff
@@ -103,25 +103,37 @@ public class PenShape: Shape, ShapeWithStrokeState {
     }
     var lastWidth = segments[0].width
     var hasStroked = false // make sure we finally stroke the path
-    for segment in (onlyLast ? [segments.last!] : segments) {
+    for (i, segment) in (onlyLast ? [segments.last!] : segments).enumerated() {
       hasStroked = false
-      context.setLineWidth(segment.width)
       let needsStroke = segment.width != lastWidth
+      context.setLineWidth(segment.width)
       if let previousMid = lastSegment?.midPoint {
         let currentMid = segment.midPoint
         context.move(to: previousMid)
         context.addQuadCurve(to: currentMid, control: segment.a)
-      } else {
+        // Usually we only draw up to the mid point of the segment, but if the
```

Comment:
What's the reasoning for only drawing up to the mid point of the segment?

Reply:
Because in the next loop iteration, we draw from the midpoint to the midpoint of the NEXT line with a nice smooth bezier curve.

Reply:
Oops, actually the other way around. If this is the END of the line, we've only just drawn up to the segment midpoint (see line 113). We need to now draw from the midpoint to the end, or the shape won't reach the final point.

Not really noticeable in practice unless you're paying close attention, but it is more correct.